activiti-cloud-service-common/activiti-cloud-services-common-identity-keycloak/src/test/java/org/activiti/cloud/services/identity/keycloak/KeycloakClientIT.java

```diff
@@ -117,6 +122,66 @@ public void should_getGroupRoles() {
         assertThat(roles).extracting("name").contains(ACTIVITI_USER_ROLE);
     }
 
+    @Test
+    public void shouldGetGroupRolesFromCache() throws InterruptedException {
+        Cache cache = cacheManager.getCache("groupRoleMapping");
+        cache.clear();
+        List<KeycloakGroup> users = keycloakClient.searchGroups("salesgroup", 0, 50);
+        String key = users.get(0).getId();
+        assertThat(cache.get(key)).isNull();
+
+        List<KeycloakRoleMapping> roles = keycloakClient.getGroupRoleMapping(key);
+
+        assertThat(roles).hasSize(1);
+        assertThat(roles).extracting("name").contains("ACTIVITI_USER");
+
+        assertThat(cache.get(key)).isNotNull();
+        Thread.sleep(5000);
```

Comment:
As written in https://rules.sonarsource.com/java/RSPEC-2925  It is better change Thread.sleep into Awaitility.await()

Reply:
Thanks, nice catch